server.js

```diff
@@ -15,6 +15,61 @@ app.get("/", function (request, response) {
 
 // TODO add your routes and helper functions here
 
-const listener = app.listen(process.env.PORT, function () {
+app.get("/booking", function (req, res) {
+  res.send(bookings);
+});
+
+app.get("/booking/:id", function (req, res) {
+  const bookingId = +req.params.id;
+  const booking = bookings.find((booking) => booking.id === bookingId);
+  if (!booking) {
+    res.status(404).send("404 error");
+  } else {
+    res.send(booking);
+  }
+});
+
+app.post("/booking", function (req, res) {
+  const createBooking = req.body;
+  createBooking.id = bookings.length + 1;
+  const {
+    title,
+    firstName,
+    surname,
+    email,
+    roomId,
+    checkInDate,
+    checkOutDate,
+  } = createBooking;
+  const valid =
+    !!title &&
+    !!firstName &&
+    !!surname &&
+    !!email &&
+    (roomId === 0 || !!roomId) &&
+    !!checkInDate &&
+    !!checkOutDate;
+  if (valid) {
+    bookings.push(createBooking);
+    res.json(bookings);
+  } else {
+    res.status(400).send("Missing Information");
```

Comment:
nice error checking, I like it. You could be more informative with what information is missing though in the response.